bash/mac

```diff
@@ -50,11 +72,15 @@ if [ $debug -eq 1 ]; then
     echo "mac chkmac=$chkmac"
 fi
 
-${dir}/mz390 $1 bal noasm $sysmac $syscpy $chksrc $chkmac $2 $3 $4 $5 $6 $7 $8 $9
+${dir}/mz390 $zFile bal noasm $sysmac $syscpy $chksrc $chkmac $2 $3 $4 $5 $6 $7 $8 $9
 rc=$?
 if [ $debug -eq 1 ]; then echo "mac: mz390 rc=$rc"; fi
-if [ "$rc" -ne "0" ]; then
-    echo "mac: mz390 error; see errors on mz390 generated $1.BAL file and console"
+if [ $rc -ne 0 ]; then
+    if [ $rc -eq 4 ]; then
+        echo "mac WARNING: mz390 rc=$rc; see warnings in mz390 generated $zFile.BAL/ERR file(s) and on console"
+    else
+        echo "mac ERROR: mz390 rc=$rc; see errors in mz390 generated $zFile.BAL/ERR file(s) and on console"
+    fi
```

Comment:
TIL: rc=0 means OK, rc=4 means warnings, rc=<any other value> means error. Personally I would prefer a range here like 0=OK,1-4=Warn, 5+=Error even though I suspect that values other than 0, 4 and some other value are the expected values.. Just a comment for thought..

Reply:
IBM has used return codes that are multiples of 4 since the System 360 days. The z390 code also does.

I think the main reason was that a return code received in a calling program could then be used to "branch to a branch table entry" via "B    BRTAB(R15)". Register 15 contains the return code (a multiple of 4), and the code at BRTAB is a sequence of branches: like "B    RC0CODE", "B     RC4CODE", etc. The appropriate code is at each of the labels. A typical use might be
```
         ...
         L     R15,=A(SUB)     Address of subroutine
         BALR  R14,R15         Call subroutine
         B     BRTAB(R15)      Continue according to RC
BRTAB    DS    0H
         B     RC0CODE         Return code 0 processing
         B     RC4CODE         Return code 4 processing
         ...
```
This technique saves the caller from having to test each return code and continue accordingly - just 2 instructions are needed.